helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -146,17 +147,39 @@
             .build());
   }
 
+  /**
+   * Determine what address to use when attempting to communicate with containers deployed via the
+   * helios-solo container. This will be passed into the helios-solo container as the HOST_ADDRESS
+   * environment variable and is later used by TemporaryJob to figure out how to reach ports mapped
+   * by the container on the host.
+   * <p>
+   * The returned value is the address of the {@code dockerHost} unless the address is determined to
+   * be localhost or 127.0.0.1.</p>
+   */
+  private String determineHeliosHost(final Info dockerInfo) throws HeliosDeploymentException {
+    // conditions where the gateway IP address given to a container should be used
+    if (dockerHostAddressIsLocalhost() && !isDockerForMac(dockerInfo)) {
```

Comment:
We don't need to determine the gateway, but could there still be value in running the probe container when using Docker for Mac?

For context, `checkDockerAndGetGateway()` used to be two separate functions with distinct responsibilities - one to test whether we can communicate with Docker from inside the container and another to determine the gateway IP from inside the container. We merged them into the same function because they span up the same Docker image twice for different reasons.